### DISCUSSION TITLE
[@types/chrome] Cannot find name chrome if declaring vitest/globals in tsconfig.json

### DISCUSSION THREAD

**Discussion by kireet3213** [DefinitelyTyped/DefinitelyTyped#71258] [Issues with a @types package] [answered]
Can someone please explain why do I have to import [@types/chrome] separately in tsconfig.json if I am using "vitest/globals"

![image](https://github.com/user-attachments/assets/e6be7d37-1498-4451-8583-f45e818933a2)

Ideally it should just work if types are available in node_modules.

Note: This is a react project and all the types for react,react-dom and node are working fine. I am using vite for bundling and vitest as a test runner

[a comment by an automated account is not shown]

**Comment by erwanjugand** [comment]
Hello @kireet3213, could you share a minimal reproduction?

**Reply by kireet3213** [reply]
@erwanjugand Can you clone this : https://github.com/kireet3213/Test

**Reply by kireet3213** [reply]
Just an update that sometimes I have to completely remove types property instead of just vitest/globals from compilerOptions. I have included that in App.tsx in the [this](https://github.com/kireet3213/Test) repo. Thanks for looking into this.

**Reply by erwanjugand** [reply, marked as answer]
https://www.typescriptlang.org/tsconfig/#types

> By default all visible ”@types” packages are included in your compilation.
> ...
> If types is specified, only packages listed will be included in the global scope.

**Reply by kireet3213** [reply]
@erwanjugand  I guess my lazy arse didn't read the docs once again.

But now I have one more question(might be foolish)- Why do the react and react-dom types work if I only specify ["vitest/globals", "chrome"] in config. According to docs only specified types are included in global scope

**Reply by erwanjugand** [reply]
Unfortunately, I don't know

**Reply by fregante** [reply]
My guess is that the files where react types appear are not properly listed in your tsconfig. I've had some strange errors before when I forgot to add new files/folders to the `includes`/`files` arrays.
